vendors/espressif/boards/esp32/ports/ble/nimble/iot_ble_hal_gatt_server.c

```diff
@@ -373,10 +381,13 @@ static int prvGATTCharAccessCb( uint16_t conn_handle,
             trans_id = ( uint32_t ) ctxt;
             if( xGattServerCb.pxRequestWriteCb != NULL )
             {
+                xSemLock = 1;
```

Comment:
Hold on. @prasad-alatkar Why is this done here? Shouldn't this be set in the `else` part of the below `if` block?